Lib9c/Action/Reregister.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Bencodex.Types;
+using Lib9c.Model.Order;
+using Libplanet;
+using Libplanet.Action;
+using Libplanet.Assets;
+using Nekoyume.Model.Item;
+using Nekoyume.Model.Mail;
+using Nekoyume.Model.State;
+using Nekoyume.TableData;
+using Serilog;
+using static Lib9c.SerializeKeys;
+using BxDictionary = Bencodex.Types.Dictionary;
+using BxList = Bencodex.Types.List;
+
+namespace Nekoyume.Action
+{
+    [Serializable]
+    [ActionType("reregister")]
+    public class Reregister : GameAction
```

Comment:
This class name need to be more specific and meaningful.🙏

Reply:
To maintain consistency with `sell`or `buy`, I made the action name only as verbs without a subject. If you have a good name, please suggest it.🙏

Reply:
I think 'Sell' and 'Buy' fall into the category of 'deal', whatever their object. And I think 'Reregister' has a broader meaning than that.
Considering we already have 'SellCancellation', how about 'ReSell'?
Of course.. I think it should have been 'RegisterToShop' rather than 'Sell' in the first place, and I wish the resale was 'ReregisterToShop'.😅

Reply:
Let's change it to `resell`. Let's change other places that use reregister as well.

Reply:
Change completed !